services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -353,7 +353,7 @@ public static Builder builder() {
    * @since 3.0.0
    */
   @SerializedName("uuid")
-  @Nullable
+  @NonNull
```

Comment:
Note that a NoRoute response will not include a `uuid` field. Not sure if that's a problem here or not.

Reply:
thanks for pointing it out.
I tried to request a route to an island without ferry but I still got a route :(

Do you have such case that will have a `NoRoute` response so I can try?

Reply:
I think it's safe because we're not parsing if there are no routes 👀 

https://github.com/mapbox/mapbox-java/blob/baef4494dcda2812569b5af676d85b45a0dbfb21/services-directions/src/main/java/com/mapbox/api/directions/v5/DirectionsResponseFactory.java#L78
1:56
https://github.com/mapbox/mapbox-java/blob/baef4494dcda2812569b5af676d85b45a0dbfb21/services-directions/src/main/java/com/mapbox/api/directions/v5/DirectionsResponseFactory.java#L44-L48

In any case, definitely worth checking. You can try `2.308538,48.852698 to 120.960515,23.69781` from https://github.com/mapbox/mapbox-java/issues/682 Also I think if you reverse the order of `lat` `lon` should also return `NoRoute`. Could you confirm @danpaz?

Reply:
You can make a request between disconnected continents, like Europe and North America.

Reply:
Just verified that in `NoRoute` case, the `uuid` is an empty string.

also, discussed with @Guardiola31337 offline, we don't parse the response if there are no routes.
see [here](https://github.com/mapbox/mapbox-java/blob/baef4494dcda2812569b5af676d85b45a0dbfb21/services-directions/src/main/java/com/mapbox/api/directions/v5/DirectionsResponseFactory.java#L78) and [here](https://github.com/mapbox/mapbox-java/blob/baef4494dcda2812569b5af676d85b45a0dbfb21/services-directions/src/main/java/com/mapbox/api/directions/v5/DirectionsResponseFactory.java#L44-L48).

We should be good here.

cc: @danpaz / @Guardiola31337

Reply:
Thanks for checking 👍  🙏